source/FreeRTOS_IP_Utils.c

```diff
@@ -1764,3 +1767,18 @@ uint16_t usChar2u16( const uint8_t * pucPtr )
              ( ( ( uint32_t ) pucPtr[ 1 ] ) ) );
 }
 /*-----------------------------------------------------------*/
+
+#if ( ipconfigUSE_DHCPv6 == 1 ) || ( ipconfigUSE_DHCP == 1 )
+
+/**
+ * @brief Returns the current state of a DHCP process.
+ *
+ * @param[in] pxEndPoint the end-point which is going through the DHCP process.
+ */
+    eDHCPState_t eGetDHCPState( const struct xNetworkEndPoint * pxEndPoint )
```

Comment:
Why is this needed to be moved out from DHCP.c?

Reply:
Its used by `xSendDHCPEvent` which is used by both DHCPv6 and DHCPv4

Reply:
Ok, in future i think we should change DHCP.c to DHCPv4.c in any major release.